### DISCUSSION TITLE
Detect if line-in is available as a source?

### DISCUSSION THREAD

**Discussion by HeraldCoupe** [hklages/node-red-contrib-sonos-plus#310] [General]
I have an occasional problem where Line-in disappears from the 'Your Sources' section of the Sonos app and the line-in functionality stops working. The only solution I know of is to reboot the speaker that the line is physically connected to. I have the speaker on a z-wave device so if I can determine the line-in has disappeared then I can restart the speaker automatically.
So, is there a command in the universal node that can list the currently available sources? I've looked through the help but haven't found anything.
Thanks in advance.

**Comment by hklages** [comment]
Hi. We have to try different options. 

Lets start with AudioIn event - see [description](https://sonos.svrooij.io/services/audio-in)
There is a boolean LineInConnected. 

Events are captured with the sister node [sonos-events](https://flows.nodered.org/node/node-red-contrib-sonos-events)
There you have to subcribe to Audio In.

Does that work for you?
